modules/alloy/src/main/ts/ephox/alloy/positioning/view/Bounder.ts

```diff
@@ -67,24 +67,25 @@ const calcReposition = (newX: number, newY: number, width: number, height: numbe
   };
 };
 
-const attempt = (candidate: SpotInfo, width: number, height: number): BounderAttemptAdt  => {
+const attempt = (candidate: SpotInfo, width: number, height: number, bounds: Boxes.Bounds): BounderAttemptAdt  => {
   const candidateX = candidate.x();
   const candidateY = candidate.y();
-  const bubbleLeft = candidate.bubble().offset().left();
-  const bubbleTop = candidate.bubble().offset().top();
+  const bubbleOffsets = candidate.bubble().offset();
+  const bubbleLeft = bubbleOffsets.left();
+  const bubbleTop = bubbleOffsets.top();
 
-  const bounds = candidate.bounds();
-  const boundsY = bounds.y();
-  const boundsBottom = bounds.bottom();
-
-  const boundsX = bounds.x();
-  const boundsRight = bounds.right();
+  // adjust the bounds to account for the layout and bubble restrictions
+  const adjustedBounds = LayoutBounds.adjustBounds(bounds, candidate.boundsRestriction(), bubbleOffsets);
+  const boundsY = adjustedBounds.y();
+  const boundsBottom = adjustedBounds.bottom();
+  const boundsX = adjustedBounds.x();
+  const boundsRight = adjustedBounds.right();
```

Comment:
Do these have to be thunked? I assume yes, but it would be a little simpler if not...

Reply:
No, but that's way outside the scope of this task as unthunking bounds is an alloy API breaking change.